ui/src/components/Layout/NavBar/index.tsx

```diff
@@ -54,12 +56,26 @@ let navBarItems: NavBarLink[] = [
 interface NavBarProps {
   open?: boolean
   setClosed: () => void
+  authEnabled: boolean
 }
 
 const NavBar: React.FC<NavBarProps> = ({ open, setClosed }) => {
   const navRef = useRef<HTMLDivElement>(null)
   const SearchCtx = useContext(SearchContext)
   const basePath = process.env.NEXT_PUBLIC_BASE_PATH ?? ''
+  const { authEnabled, isAuthenticated } = useAuth()
+
+  const handleLogout = async () => {
+    try {
+      await fetch('/api/authentication/logout', {
+        method: 'POST',
+        credentials: 'include',
+      })
+      window.location.href = '/login'
+    } catch (error) {
+      console.error('Failed to log out:', error)
+    }
+  }
```

Comment:
Given that logout is just removing 2 cookies, this can all be done client side in JavaScript, I think you've got this somewhere already.